tools/ARIAtools/tsSetup.py

```diff
@@ -205,52 +238,46 @@ def generate_stack(aria_prod, input_files, output_file_name,
 
     # Set up single stack file
     if not os.path.exists(os.path.join(workdir, 'stack')):
-        print('Creating directory: {os.path.join(workdir, "stack")}')
+        print('Creating directory: ', os.path.join(workdir, "stack"))
         os.makedirs(os.path.join(workdir, 'stack'))
-    else:
-        print(f'Directory {os.path.join(workdir, "stack")} already exists.')
 
-    if input_files in ['unwrappedPhase', 'unwrapped', 'unw']:
+    # MG; did a little clean up below
+
+    # MG: find pattern in string, with special case for tropo
+    # take into account variation in input name
+    # e.g. [unwrappedPhase, unwrapPhase, unwrap]
+    # NOTE is it necessary to have the option for different names
+    # as they are hardcoreded, maybe because of different nc version, double-check?
+    _find_match = lambda x: re.findall('[A-Z][^A-Z]*', x)[0] \
+                            if re.search('tropo', x) \
+                            else re.findall('[a-z][^A-Z]*', x)[0]
+
+    # Special case - gacos correction files
```

Comment:
Do not know Sim is it better to move older version of code for correcting unwrapped Phase with total tropo delay from GACOS to separate script:
1.  ingest GACOS products, and prepare tropoTotal vrt files (in a similar manner like ariaExtract.py)
2. use tsSetup.py to prepare tropoTotal stack vrt
3. have optional script to treat unwrappedPhase files with GACOS/other model tropoTotal files

Reply:
This is simplify both ariaExtract and tsSetup scripts,

Reply:
@mgovorcin I understood that the intention was though to no longer directly correct the phase through ARIA-tools. I.e. once I merge these changes, we should work to now push corrections to a separate cube which we then interface with MintPy to use instead to correct the phase on their end.